src/common/hook/useQueue/utils/publish.tsx

```diff
@@ -143,19 +163,24 @@ export const groupDocumentsIntoJobs = async (
   }
 
   // Process all transferable records next
-  transferableRecords.forEach((transferableRecord) => {
+  for (const transferableRecord of transferableRecords) {
     const { type, contractAddress, rawDocument, payload } = transferableRecord;
-    const documents = wrapDocuments([rawDocument]);
+
+    const documents = utils.isRawV3Document(rawDocument)
+      ? await wrapDocumentsV3([rawDocument])
+      : wrapDocuments([rawDocument]);
```

Comment:
you've used this pattern thrice already, might be good to make it into a simple helper function `wrapDocuments`, can alias the previous `wrapDocuments` to `wrapDocumentsV2`.

Reply:
okay extracted into local helper function